ibis/expr/operations/core.py

```diff
@@ -95,37 +93,6 @@ def _pp(x):
 
         return '{}({})'.format(opname, ', '.join(pprint_args))
 
-    def __getstate__(self) -> Dict[str, AnyT]:
-        """The attributes _expr_cached and _hash are
-        used as caches; they can be excluded from
-        serialization without affecting correctness.
-
-        Excluding _expr_cached and _hash from serialization
-        will allow the serialized bytes to be the same for
-        equivalent Node objets.
-
-        Returns
-        -------
-        Dict[str, Any]
-            A dictionary storing the objects attributes.
-        """
-        excluded_slots = {'_expr_cached', '_hash'}
-        return {
-            slot: getattr(self, slot)
-            for slot in self.__slots__
```

Comment:
`__slots__` doesn't store the inherited keys anymore, so we need to use `self.__signature__.keys()`. Since this is a useful feature for `Annotable` I moved this implementation there.